Mathlib/Probability/Distributions/Gaussian.lean

```diff
@@ -308,19 +308,26 @@ variable {Ω : Type} [MeasureSpace Ω]
 
 /-- If `X` is a real random variable with Gaussian law with mean `μ` and variance `v`, then `X + y`
 has Gaussian law with mean `μ + y` and variance `v`. -/
-lemma gaussianReal_add_const {X : Ω → ℝ} (hX : Measure.map X ℙ = gaussianReal μ v)
-    (hXm : Measurable X) (y : ℝ) :
+lemma gaussianReal_add_const {X : Ω → ℝ} (hX : Measure.map X ℙ = gaussianReal μ v) (y : ℝ) :
     Measure.map (fun ω ↦ X ω + y) ℙ = gaussianReal (μ + y) v := by
+  have hXm : AEMeasurable X := by
+    by_contra h
+    rw [Measure.map_of_not_aemeasurable h] at hX
+    exact absurd hX.symm (inferInstance : NeZero _).out
   change Measure.map ((fun ω ↦ ω + y) ∘ X) ℙ = gaussianReal (μ + y) v
-  rw [← Measure.map_map (measurable_id'.add_const _) hXm, hX, gaussianReal_map_add_const y]
+  rw [← AEMeasurable.map_map_of_aemeasurable (measurable_id'.add_const _).aemeasurable hXm, hX,
+    gaussianReal_map_add_const y]
 
 /-- If `X` is a real random variable with Gaussian law with mean `μ` and variance `v`, then `c * X`
 has Gaussian law with mean `c * μ` and variance `c^2 * v`. -/
-lemma gaussianReal_const_mul {X : Ω → ℝ} (hX : Measure.map X ℙ = gaussianReal μ v)
-    (hXm : Measurable X) (c : ℝ) :
+lemma gaussianReal_const_mul {X : Ω → ℝ} (hX : Measure.map X ℙ = gaussianReal μ v) (c : ℝ) :
     Measure.map (fun ω ↦ c * X ω) ℙ = gaussianReal (c * μ) (⟨c^2, sq_nonneg _⟩ * v) := by
+  have hXm : AEMeasurable X := by
```

Comment:
maybe register a lemma `[NeZero μ] (h : Measure.map f nu = μ) : AEMeasurable f nu`?

Reply:
I added these:
```lean
@[simp] lemma neZero_zero_iff_false {α : Type*} [Zero α] : NeZero (0 : α) ↔ False :=

lemma aemeasurable_of_map_neZero {mβ : MeasurableSpace β} {μ : Measure α}
    {f : α → β} (h : NeZero (μ.map f)) :
    AEMeasurable f μ := by
```
and added `@[simp]` on `map_of_not_aemeasurable`